### PR TITLE
Add internal functions for converting words to ext rep obj words and to strings

### PR DESCRIPTION
**Pull request by james-d-mitchell**
The internal functions `SEMIGROUPS.WordToExtRepObj`,
`SEMIGROUPS.ExtRepObjToString`, and `SEMIGROUPS.WordToString` are introduced.

### CODE REVIEW

File: gap/semigroups/semifp.gi [hunk 1]
```diff
@@ -464,6 +464,59 @@ SEMIGROUPS.ExtRepObjToWord := function(ext_rep_obj)
...
+SEMIGROUPS.WordToString := function(word)
+  local alphabet, out, letter;
+  alphabet := "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
+  out := "";
+  for letter in word do
+    if letter > Length(alphabet) then
+      ErrorNoReturn("SEMIGROUPS.WordToString: the maximum value in the",
+                    " argument be at most ", Length(alphabet), ",");
+    fi;
+    Append(out, [alphabet[letter]]);
```
Comment (wilfwilson) [inline gap/semigroups/semifp.gi]: This should be `Add(out, alphabet[letter])`.

File: gap/semigroups/semifp.gi [hunk 2]
```diff
@@ -464,6 +464,59 @@ SEMIGROUPS.ExtRepObjToWord := function(ext_rep_obj)
...
+  local alphabet, out, i;
+  alphabet := "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
+  out := "";
+  for i in [1, 3 .. Length(ext_rep_obj) - 1] do
+    if ext_rep_obj[i] > Length(alphabet) then
+      ErrorNoReturn("SEMIGROUPS.ExtRepObjToString: the maximum value in an ",
+                    "odd position of the argument must be at most ",
+                    Length(alphabet), ",");
+    fi;
+    Append(out, [alphabet[ext_rep_obj[i]]]);
```
Comment (wilfwilson) [inline gap/semigroups/semifp.gi]: Here you may as well also do `Add(out, alphabet[ext_rep_obj[i]]);`

**Comment by james-d-mitchell** [comment]
Done @wil

**Comment by mtorpey** [comment]
Looks good to me too.  Since you've addressed Wilf's concerns, I'll merge.

**Pull request merged**